### PR TITLE
[String] Add support for kebab case

### LINKED ISSUE #48780: [String] Add `kebab()` to methods to change case
### Description

Hi,

It would be interesting to add the `kebab()` function to the string component. 

### Example

```php
u('FOO Bar')->kebab();     // 'foo-bar'
```

### PR DESCRIPTION
**Pull request by seb-jean**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #48780
| License       | MIT
| Doc PR        | n/a

I add suport for kebab case for String Component.

### CODE REVIEW

File: src/Symfony/Component/String/AbstractUnicodeString.php [hunk 1]
```diff
@@ -214,6 +214,14 @@ public function join(array $strings, string $lastGlue = null): static
         return $str;
     }
 
+    public function kebab(): static
+    {
+        $str = $this->camel();
```
Comment (OskarStark) [inline src/Symfony/Component/String/AbstractUnicodeString.php:219]: camel()->lower()->replace('_', '-')

Is not enough? Just asking

File: src/Symfony/Component/String/AbstractString.php [hunk 2]
```diff
@@ -390,6 +390,8 @@ public function jsonSerialize(): string
         return $this->string;
     }
 
+    abstract public function kebab(): static;
```
Comment (derrabus) [inline src/Symfony/Component/String/AbstractString.php:393]: `AbstractString` is not `@internal`, so this is a BC-break, I believe.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/String/AbstractUnicodeString.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/String/AbstractUnicodeString.php:219]: If that's the case, maybe we don't need yet another method?

Comment (seb-jean) [reply src/Symfony/Component/String/AbstractUnicodeString.php:219]: The `snake()` function uses the same process:
https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/String/AbstractUnicodeString.php#L349

Comment (seb-jean) [reply src/Symfony/Component/String/AbstractUnicodeString.php:219]: > camel()->lower()->replace('_', '-')
> 
> Is not enough? Just asking

Your example does not work.

Comment (jdreesen) [reply src/Symfony/Component/String/AbstractUnicodeString.php:219]: Right, it's `snake()->replace('_', '-')`, I think.

**Pull request closed without merging**